services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -150,13 +160,130 @@ class PortHistorySpec extends HealthCheckSpecification {
         }
     }
 
-    void checkPortHistory(PortHistoryResponse portHistory, SwitchId switchId, Integer port, String event) {
+    def "Port history is able to show ANTI_FLAP statistic"() {
+        given: "A direct link"
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+        assumeTrue("Unable to find ISL for this test", isl as boolean)
+        def timestampBefore = System.currentTimeMillis()
+
+        when: "Execute port DOWN on the src switch"
+        antiflap.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET / 2) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        then: "Port history is created on the src switch"
+        Wrappers.wait(WAIT_OFFSET) {
+            Long timestampAfterDown = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+                assert it.size() == 2
+            }
+        }
+
+        when: "Generate antiflap statistic"
+        def count = 0
+        //confd/templates/wfm/topology.properties.tmpl =>  port.antiflap.stats.dumping.interval.seconds = 60
+        // this variable means how often the 'ANTI_FLAP_PERIODIC_STATS' will be logged in port history
+        def antiflapDumpingInterval = 60
+        Wrappers.timedLoop(antiflapDumpingInterval * 0.9) {
+            northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+            sleep(2000)
+            northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+            sleep(2000)
+            count += 1
+        }
+
+        then: "Antiflap statistic is available in port history"
+        Wrappers.wait(antiflapDumpingInterval * 0.1 + WAIT_OFFSET) {
+            Long timestampAfterStat = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterStat)) {
+                assert it.size() == 3
+                def antiflapStat = it[-1]
+                checkPortHistory(antiflapStat, isl.srcSwitch.dpId, isl.srcPort,
+                        PortHistoryEvent.ANTI_FLAP_PERIODIC_STATS)
+                assert count == antiflapStat.downCount
+                assert count == antiflapStat.upCount
+            }
+        }
+
+        and: "Cleanup: revert system to original state"
+        antiflap.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval + antiflapCooldown) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+    }
+
+    def "System shows antiflap statistic in the ANTI_FLAP_DEACTIVATED event when antiflap is deactivated\
+ before collecting ANTI_FLAP_PERIODIC_STATS"() {
+        given: "A direct link"
+        assumeTrue("antiflap.cooldown should be three times less than port.antiflap.stats.dumping.interval.seconds",
```

Comment:
Also I don't really understand what is behind this logic

Reply:
The aim of this test is: check that system collects stats in `ANTI_FLAP_DEACTIVATED` event when `ANTI_FLAP_PERIODIC_STATS` was never executed.
it can be done when `antiflapCooldown` is three times less than `port.antiflap.stats.dumping.interval.seconds`

Reply:
aaaand how did you calculate the '3 times less' statement? My point of view: you just invented this rule on fly and it is not real. The real limitation is: `cooldown + durationYouFlap < ANTI_FLAP_DEACTIVATED`. I don't see where the '3 times' comes from

Reply:
fixed totally